source/code/plugin/kubernetes_container_inventory.rb

```diff
@@ -203,7 +203,7 @@ def obtainContainerEnvironmentVars(containerId)
               envVars = File.read(environFilePath, 200000)             
               if !envVars.nil? && !envVars.empty?
                 envVars = envVars.split("\0")
-                envValueString = envVars.to_s
+                envValueString = envVars.join('')
```

Comment:
Not sure  whats the problem with to_s, in this case, envars will be string array ["a1=b1","a2=b2"] and to_s will be converting to string i.e. '["a1=b1","a2=b2"]' where as join will result "a1=b1" "a2=b2" which is not JSON string at all. This is going to break the JSON parsing functionality completely.   Please help me to understand what is the issue with to_s at all. https://blog.bigbinary.com/2012/06/26/to_str-in-ruby.html

Reply:
@ganga1980  This is a draft PR - its not ready for review yet. I know the current commited code isn't a JSON string. It was something I was playing around with.

When you try using to_s on an array in ruby you get extra escaped characters.
PS C:\Users\kadubey\Documents\temp> cat .\hello.rb                                                                                                                                                                                                     temp = ["test=#@$#24123das"]

puts temp.to_s;
PS C:\Users\kadubey\Documents\temp> ruby .\hello.rb                                                                                                                                                                                                    ["test=\#@$#24123das"]

The extra \ being added in front of the # is wrong.

Reply:
Updated to use to_json to do the conversion. This does not add extra escape characters and as the function name suggets its also a valid json string which matches the behavior that the current and older agents exhibit.

Reply:
This PR is ready for review now @ganga1980